ts/util/accountManager.ts

```diff
@@ -302,21 +301,31 @@ export async function deleteEverythingAndNetworkData() {
 
     // clear all sogs inboxes (includes message requests)
     const allRoomInfos = await getAllValidOpenGroupV2ConversationRoomInfos();
-    if (allRoomInfos && allRoomInfos.size > 0) {
+    const allRoomInfosArray = Array.from(allRoomInfos?.values() || []);
+
+    if (allRoomInfosArray.length) {
       // clear each inbox per sogs
-      // eslint-disable-next-line no-restricted-syntax
-      for (const roomInfo of allRoomInfos.values()) {
+
+      const clearInboxPromises = allRoomInfosArray.map(async roomInfo => {
         try {
-          // eslint-disable-next-line no-await-in-loop
           const success = await clearInbox(roomInfo);
           if (!success) {
             throw Error(`Failed to clear inbox for ${roomInfo.conversationId}`);
           }
+          return { status: 'fulfilled' };
         } catch (error) {
-          window.log.info('DeleteAccount =>', error);
-          continue;
+          return { status: 'rejected', reason: error };
+        }
+      });
+
+      const results = await Promise.allSettled(clearInboxPromises);
+      results.forEach((result, index) => {
+        if (result.status === 'rejected') {
+          window.log.error(result.reason);
+        } else {
+          window.log.info('Inbox cleared for room', allRoomInfosArray[index]);
         }
-      }
+      });
```

Comment:
you shouldn't have the return {status: fullfilled/rejected by hand). You should just have to return something/throw in the promise, and Promise.allSettled will report the ones which failed as status === "rejected" and others as fullfilled.

Reply:
Fixed!